### PR TITLE
feat(sdk): Add reference data for the SDK input/output types

### PR DESCRIPTION
**Pull request by erictuvesson**
## What kind of change does this PR introduce?

Adding more insight into how the SDK works and what features are available.

I am just starting the PR right now, will keep adding to this and verify that everything is correct before merging.

### CODE REVIEW

File: sdk/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+## Getting started
+
+We have a lot of open source modules in this [GitHub repository](https://github.com/noodlapp/modules).
+These are a great place to start when looking for references, and examples of how to build a module.
+
+Not all of these modules are made with the SDK. Why?
+The SDK is built on top of the internal Noodl API, designed to make it easier and future compatible with internal Noodl API changes in mind.
+Currently the SDK does lack some of the functionality that the internal Noodl API provides, which is the primary reason not to use the SDK for these modules.
+In the end the module code will be very similar, but can cause some confusion.
```
[the remark this answers is not in the document]
Comment (erictuvesson) [reply sdk/overview.mdx:23]: I just looked over all the modules, the only one I can see that is not using the SDK is the `simple-tooltips` module. Thanks for the note, will update the text a little and update the module README.

[the remark this answers is not in the document]
Comment (trevorgildersleeve) [reply sdk/overview.mdx:23]: But even simple tooltips references the SDK?

https://github.com/noodlapp/modules/blob/13438119f29bc2c793ce38421820bc21b28b4e16/modules/simple-tooltips/package.json#L14C1-L14C1

https://github.com/noodlapp/modules/blob/13438119f29bc2c793ce38421820bc21b28b4e16/modules/simple-tooltips/src/index.js#L1

Sorry, not trying to be a pain, I'm just really trying to understand the modules.

[the remark this answers is not in the document]
Comment (erictuvesson) [reply sdk/overview.mdx:23]: A very valid question and something that would be excellent to cover.

The only place the SDK is used in the module is with `defineModule` which is calling calling `Noodl.defineModule`.

`Noodl.defineModule` is actually not defined by the SDK. It is defined by the Noodl Runtime Viewer at page load globally on the window. Like this:
```js
window.Noodl = {
  defineModule(module) {
    window.__noodl_modules.push(module);
  },
  // ...
}
```

[the remark this answers is not in the document]
Comment (trevorgildersleeve) [reply sdk/overview.mdx:23]: Ah, okay, thank you!

**Comment by erictuvesson** [comment]
@trevorgildersleeve I made some changes to the text, will merge this to keep it moving forward. Updates to the text is always welcome!

**Pull request merged**